### PR TITLE
[Fix #11255] Fix an error for `Style/RequireOrder` when `require` with no arguments is put between `require`

### LINKED ISSUE #11255: `An error occurred while Style/RequireOrder cop was inspecting /Users/jonathonfrisby/fb/app/Rakefile:4:0`
## Expected behavior

I expect RuboCop to process my code without error, as 1.39.0 did.

## Actual behavior

Describe here what actually happened.

```
For /Users/jonathonfrisby/fb/app: configuration from /Users/jonathonfrisby/fb/app/.rubocop.yml
configuration from /Users/jonathonfrisby/.rbenv/versions/3.0.4/gemsets/fasterbetter-app/gems/rubocop-eightyfourcodes-0.0.2/config/default.yml
configuration from /Users/jonathonfrisby/.rbenv/versions/3.0.4/gemsets/fasterbetter-app/gems/rubocop-eightyfourcodes-0.0.2/config/default.yml
Default configuration from /Users/jonathonfrisby/.rbenv/versions/3.0.4/gemsets/fasterbetter-app/gems/rubocop-1.40.0/config/default.yml
configuration from /Users/jonathonfrisby/.rbenv/versions/3.0.4/gemsets/fasterbetter-app/gems/rubocop-performance-1.15.1/config/default.yml
configuration from /Users/jonathonfrisby/.rbenv/versions/3.0.4/gemsets/fasterbetter-app/gems/rubocop-performance-1.15.1/config/default.yml
configuration from /Users/jonathonfrisby/.rbenv/versions/3.0.4/gemsets/fasterbetter-app/gems/rubocop-rake-0.6.0/config/default.yml
configuration from /Users/jonathonfrisby/.rbenv/versions/3.0.4/gemsets/fasterbetter-app/gems/rubocop-rake-0.6.0/config/default.yml
configuration from /Users/jonathonfrisby/.rbenv/versions/3.0.4/gemsets/fasterbetter-app/gems/rubocop-rspec-2.15.0/config/default.yml
configuration from /Users/jonathonfrisby/.rbenv/versions/3.0.4/gemsets/fasterbetter-app/gems/rubocop-rspec-2.15.0/config/default.yml
configuration from /Users/jonathonfrisby/.rbenv/versions/3.0.4/gemsets/fasterbetter-app/gems/rubocop-rubycw-0.1.6/config/default.yml
configuration from /Users/jonathonfrisby/.rbenv/versions/3.0.4/gemsets/fasterbetter-app/gems/rubocop-rubycw-0.1.6/config/default.yml
configuration from /Users/jonathonfrisby/.rbenv/versions/3.0.4/gemsets/fasterbetter-app/gems/rubocop-thread_safety-0.4.4/config/default.yml
configuration from #{path}
Use parallel by default.
Skipping parallel inspection: only a single file needs inspection
Inspecting 1 file
Scanning /Users/jonathonfrisby/fb/app/Rakefile
An error occurred while Style/RequireOrder cop was inspecting /Users/jonathonfrisby/fb/app/Rakefile:4:0.
undefined method `source' for nil:NilClass
/Users/jonathonfrisby/.rbenv/versions/3.0.4/gemsets/fasterbetter-app/gems/rubocop-1.40.0/lib/rubocop/cop/style/require_order.rb:70:in `block in find_previous_older_sibling'
/Users/jonathonfrisby/.rbenv/versions/3.0.4/gemsets/fasterbetter-app/gems/rubocop-1.40.0/lib/rubocop/cop/style/require_order.rb:65:in `each'
/Users/jonathonfrisby/.rbenv/versions/3.0.4/gemsets/fasterbetter-app/gems/rubocop-1.40.0/lib/rubocop/cop/style/require_order.rb:65:in `find'
/Users/jonathonfrisby/.rbenv/versions/3.0.4/gemsets/fasterbetter-app/gems/rubocop-1.40.0/lib/rubocop/cop/style/require_order.rb:65:in `find_previous_older_sibling'
/Users/jonathonfrisby/.rbenv/versions/3.0.4/gemsets/fasterbetter-app/gems/rubocop-1.40.0/lib/rubocop/cop/style/require_order.rb:47:in `on_send'
/Users/jonathonfrisby/.rbenv/versions/3.0.4/gemsets/fasterbetter-app/gems/rubocop-1.40.0/lib/rubocop/cop/commissioner.rb:136:in `public_send'
/Users/jonathonfrisby/.rbenv/versions/3.0.4/gemsets/fasterbetter-app/gems/rubocop-1.40.0/lib/rubocop/cop/commissioner.rb:136:in `block (2 levels) in trigger_restricted_cops'
/Users/jonathonfrisby/.rbenv/versions/3.0.4/gemsets/fasterbetter-app/gems/rubocop-1.40.0/lib/rubocop/cop/commissioner.rb:160:in `with_cop_error_handling'
/Users/jonathonfrisby/.rbenv/versions/3.0.4/gemsets/fasterbetter-app/gems/rubocop-1.40.0/lib/rubocop/cop/commissioner.rb:135:in `block in trigger_restricted_cops'
/Users/jonathonfrisby/.rbenv/versions/3.0.4/gemsets/fasterbetter-app/gems/rubocop-1.40.0/lib/rubocop/cop/commissioner.rb:134:in `each'
/Users/jonathonfrisby/.rbenv/versions/3.0.4/gemsets/fasterbetter-app/gems/rubocop-1.40.0/lib/rubocop/cop/commissioner.rb:134:in `trigger_restricted_cops'
/Users/jonathonfrisby/.rbenv/versions/3.0.4/gemsets/fasterbetter-app/gems/rubocop-1.40.0/lib/rubocop/cop/commissioner.rb:70:in `on_send'
/Users/jonathonfrisby/.rbenv/versions/3.0.4/gemsets/fasterbetter-app/gems/rubocop-ast-1.24.0/lib/rubocop/ast/traversal.rb:137:in `block in on_dstr'
/Users/jonathonfrisby/.rbenv/versions/3.0.4/gemsets/fasterbetter-app/gems/rubocop-ast-1.24.0/lib/rubocop/ast/traversal.rb:137:in `each'
/Users/jonathonfrisby/.rbenv/versions/3.0.4/gemsets/fasterbetter-app/gems/rubocop-ast-1.24.0/lib/rubocop/ast/traversal.rb:137:in `on_dstr'
/Users/jonathonfrisby/.rbenv/versions/3.0.4/gemsets/fasterbetter-app/gems/rubocop-1.40.0/lib/rubocop/cop/commissioner.rb:71:in `on_begin'
/Users/jonathonfrisby/.rbenv/versions/3.0.4/gemsets/fasterbetter-app/gems/rubocop-ast-1.24.0/lib/rubocop/ast/traversal.rb:20:in `walk'
/Users/jonathonfrisby/.rbenv/versions/3.0.4/gemsets/fasterbetter-app/gems/rubocop-1.40.0/lib/rubocop/cop/commissioner.rb:86:in `investigate'
/Users/jonathonfrisby/.rbenv/versions/3.0.4/gemsets/fasterbetter-app/gems/rubocop-1.40.0/lib/rubocop/cop/team.rb:154:in `investigate_partial'
/Users/jonathonfrisby/.rbenv/versions/3.0.4/gemsets/fasterbetter-app/gems/rubocop-1.40.0/lib/rubocop/cop/team.rb:82:in `investigate'
/Users/jonathonfrisby/.rbenv/versions/3.0.4/gemsets/fasterbetter-app/gems/rubocop-1.40.0/lib/rubocop/runner.rb:315:in `inspect_file'
/Users/jonathonfrisby/.rbenv/versions/3.0.4/gemsets/fasterbetter-app/gems/rubocop-1.40.0/lib/rubocop/runner.rb:259:in `block in do_inspection_loop'
/Users/jonathonfrisby/.rbenv/versions/3.0.4/gemsets/fasterbetter-app/gems/rubocop-1.40.0/lib/rubocop/runner.rb:293:in `block in iterate_until_no_changes'
/Users/jonathonfrisby/.rbenv/versions/3.0.4/gemsets/fasterbetter-app/gems/rubocop-1.40.0/lib/rubocop/runner.rb:286:in `loop'
/Users/jonathonfrisby/.rbenv/versions/3.0.4/gemsets/fasterbetter-app/gems/rubocop-1.40.0/lib/rubocop/runner.rb:286:in `iterate_until_no_changes'
/Users/jonathonfrisby/.rbenv/versions/3.0.4/gemsets/fasterbetter-app/gems/rubocop-1.40.0/lib/rubocop/runner.rb:255:in `do_inspection_loop'
/Users/jonathonfrisby/.rbenv/versions/3.0.4/gemsets/fasterbetter-app/gems/rubocop-1.40.0/lib/rubocop/runner.rb:138:in `block in file_offenses'
/Users/jonathonfrisby/.rbenv/versions/3.0.4/gemsets/fasterbetter-app/gems/rubocop-1.40.0/lib/rubocop/runner.rb:163:in `file_offense_cache'
/Users/jonathonfrisby/.rbenv/versions/3.0.4/gemsets/fasterbetter-app/gems/rubocop-1.40.0/lib/rubocop/runner.rb:137:in `file_offenses'
/Users/jonathonfrisby/.rbenv/versions/3.0.4/gemsets/fasterbetter-app/gems/rubocop-1.40.0/lib/rubocop/runner.rb:128:in `process_file'
/Users/jonathonfrisby/.rbenv/versions/3.0.4/gemsets/fasterbetter-app/gems/rubocop-1.40.0/lib/rubocop/runner.rb:109:in `block in each_inspected_file'
/Users/jonathonfrisby/.rbenv/versions/3.0.4/gemsets/fasterbetter-app/gems/rubocop-1.40.0/lib/rubocop/runner.rb:108:in `each'
/Users/jonathonfrisby/.rbenv/versions/3.0.4/gemsets/fasterbetter-app/gems/rubocop-1.40.0/lib/rubocop/runner.rb:108:in `reduce'
/Users/jonathonfrisby/.rbenv/versions/3.0.4/gemsets/fasterbetter-app/gems/rubocop-1.40.0/lib/rubocop/runner.rb:108:in `each_inspected_file'
/Users/jonathonfrisby/.rbenv/versions/3.0.4/gemsets/fasterbetter-app/gems/rubocop-1.40.0/lib/rubocop/runner.rb:94:in `inspect_files'
/Users/jonathonfrisby/.rbenv/versions/3.0.4/gemsets/fasterbetter-app/gems/rubocop-1.40.0/lib/rubocop/runner.rb:47:in `run'
/Users/jonathonfrisby/.rbenv/versions/3.0.4/gemsets/fasterbetter-app/gems/rubocop-1.40.0/lib/rubocop/cli/command/execute_runner.rb:26:in `block in execute_runner'
/Users/jonathonfrisby/.rbenv/versions/3.0.4/gemsets/fasterbetter-app/gems/rubocop-1.40.0/lib/rubocop/cli/command/execute_runner.rb:52:in `with_redirect'
/Users/jonathonfrisby/.rbenv/versions/3.0.4/gemsets/fasterbetter-app/gems/rubocop-1.40.0/lib/rubocop/cli/command/execute_runner.rb:25:in `execute_runner'
/Users/jonathonfrisby/.rbenv/versions/3.0.4/gemsets/fasterbetter-app/gems/rubocop-1.40.0/lib/rubocop/cli/command/execute_runner.rb:17:in `run'
/Users/jonathonfrisby/.rbenv/versions/3.0.4/gemsets/fasterbetter-app/gems/rubocop-1.40.0/lib/rubocop/cli/command.rb:11:in `run'
/Users/jonathonfrisby/.rbenv/versions/3.0.4/gemsets/fasterbetter-app/gems/rubocop-1.40.0/lib/rubocop/cli/environment.rb:18:in `run'
/Users/jonathonfrisby/.rbenv/versions/3.0.4/gemsets/fasterbetter-app/gems/rubocop-1.40.0/lib/rubocop/cli.rb:72:in `run_command'
/Users/jonathonfrisby/.rbenv/versions/3.0.4/gemsets/fasterbetter-app/gems/rubocop-1.40.0/lib/rubocop/cli.rb:79:in `execute_runners'
/Users/jonathonfrisby/.rbenv/versions/3.0.4/gemsets/fasterbetter-app/gems/rubocop-1.40.0/lib/rubocop/cli.rb:48:in `run'
/Users/jonathonfrisby/.rbenv/versions/3.0.4/gemsets/fasterbetter-app/gems/rubocop-1.40.0/exe/rubocop:19:in `block in <top (required)>'
/Users/jonathonfrisby/.rbenv/versions/3.0.4/lib/ruby/3.0.0/benchmark.rb:308:in `realtime'
/Users/jonathonfrisby/.rbenv/versions/3.0.4/gemsets/fasterbetter-app/gems/rubocop-1.40.0/exe/rubocop:19:in `<top (required)>'
/Users/jonathonfrisby/.rbenv/versions/3.0.4/gemsets/fasterbetter-app/bin/rubocop:25:in `load'
/Users/jonathonfrisby/.rbenv/versions/3.0.4/gemsets/fasterbetter-app/bin/rubocop:25:in `<main>'
.

1 file inspected, no offenses detected

1 error occurred:
An error occurred while Style/RequireOrder cop was inspecting /Users/jonathonfrisby/fb/app/Rakefile:4:0.
Errors are usually caused by RuboCop bugs.
Please, report your problems to RuboCop's issue tracker.
https://github.com/rubocop/rubocop/issues

Mention the following information in the issue report:
1.40.0 (using Parser 3.1.3.0, rubocop-ast 1.24.0, running on ruby 3.0.4) [arm64-darwin21]
Finished in 0.25962700007949024 seconds
```

## Steps to reproduce the problem

The problematic line in Rakefile is this:

```ruby
Bundler.require
```

Changing that to have an argument (e.g. `Bundler.require(:development)`) fixes the error.

## RuboCop version

```
% rubocop -V
1.40.0 (using Parser 3.1.3.0, rubocop-ast 1.24.0, running on ruby 3.0.4) [arm64-darwin21]
  - rubocop-performance 1.15.1
  - rubocop-rake 0.6.0
  - rubocop-rspec 2.15.0
  - rubocop-rubycw 0.1.6
  - rubocop-thread_safety 0.4.4
```

### PR DESCRIPTION
**Pull request by ydah**
Fix: #11255

-----------------

Before submitting the PR make sure the following are checked:

* [x] The PR relates to *only* one subject with a clear title and description in grammatically correct, complete sentences.
* [x] Wrote [good commit messages][1].
* [x] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [x] Squashed related commits together.
* [x] Added tests.
* [x] Ran `bundle exec rake default`. It executes all tests and runs RuboCop on its own code.
* [x] Added an entry (file) to the [changelog folder](https://github.com/rubocop/rubocop/blob/master/changelog/) named `{change_type}_{change_description}.md` if the new code introduces user-observable changes. See [changelog entry format](https://github.com/rubocop/rubocop/blob/master/CONTRIBUTING.md#changelog-entry-format) for details.

[1]: https://chris.beams.io/posts/git-commit/

### CODE REVIEW

File: lib/rubocop/cop/style/require_order.rb [hunk 1]
```diff
@@ -63,14 +65,21 @@ def on_send(node)
...
         end
 
+        def sortable_sibling?(node, sibling)
+          sibling.respond_to?(:send_type?) &&
+            sibling.send_type? &&
+            sibling.method?(node.method_name) &&
+            sibling.arguments? &&
+            !sibling.receiver &&
+            in_same_section?(sibling, node)
+        end
```
Comment (koic) [inline lib/rubocop/cop/style/require_order.rb:81]: It seems possible to inline by removing unnecessary conditions as follows.

```diff
diff --git a/lib/rubocop/cop/style/require_order.rb b/lib/rubocop/cop/style/require_order.rb
index d071472a9..12ecc1f7f 100644
--- a/lib/rubocop/cop/style/require_order.rb
+++ b/lib/rubocop/cop/style/require_order.rb
@@ -65,21 +65,14 @@ module RuboCop

         def find_previous_older_sibling(node)
           node.left_siblings.reverse.find do |sibling|
-            break unless sortable_sibling?(node, sibling)
+            break unless sibling.method?(node.method_name)
+            break unless sibling.arguments? && !sibling.receiver
+            break unless in_same_section?(sibling, node)

             node.first_argument.source < sibling.first_argument.source
           end
         end

-        def sortable_sibling?(node, sibling)
-          sibling.respond_to?(:send_type?) &&
-            sibling.send_type? &&
-            sibling.method?(node.method_name) &&
-            sibling.arguments? &&
-            !sibling.receiver &&
-            in_same_section?(sibling, node)
-        end
-
         def in_same_section?(node1, node2)
           !node1.location.expression.with(
             end_pos: node2.location.expression.end_pos
```

**Comment by koic** [comment]
Can you update with the following patch to prevent CI error:

```diff
diff --git a/lib/rubocop/cop/style/require_order.rb b/lib/rubocop/cop/style/require_order.rb
index 12ecc1f7f..952f533eb 100644
--- a/lib/rubocop/cop/style/require_order.rb
+++ b/lib/rubocop/cop/style/require_order.rb
@@ -65,7 +65,7 @@ module RuboCop

         def find_previous_older_sibling(node)
           node.left_siblings.reverse.find do |sibling|
-            break unless sibling.method?(node.method_name)
+            break unless sibling.send_type? && sibling.method?(node.method_name)
             break unless sibling.arguments? && !sibling.receiver
             break unless in_same_section?(sibling, node)

diff --git a/spec/rubocop/cop/style/require_order_spec.rb b/spec/rubocop/cop/style/require_order_spec.rb
index 89c5cade9..81632aac1 100644
--- a/spec/rubocop/cop/style/require_order_spec.rb
+++ b/spec/rubocop/cop/style/require_order_spec.rb
@@ -130,4 +130,15 @@ RSpec.describe RuboCop::Cop::Style::RequireOrder, :config do
       RUBY
     end
   end
+
+  context 'when something other than a method call is used between `require`' do
+    it 'registers no offense' do
+      expect_no_offenses(<<~RUBY)
+        require 'a'
+        begin
+        end
+        require 'b'
+      RUBY
+    end
+  end
 end
```

```console
An error occurred while Style/RequireOrder cop was inspecting /Users/koic/src/github.com/rubocop/rubocop/Rakefile:20:0.
undefined method `method?' for s(:kwbegin,
  s(:rescue,
    s(:send,
      s(:const, nil, :Bundler), :setup,
      s(:sym, :default),
      s(:sym, :development)),
    s(:resbody,
      s(:array,
        s(:const,
          s(:const, nil, :Bundler), :BundlerError)),
      s(:lvasgn, :e),
      s(:begin,
        s(:send, nil, :warn,
          s(:send,
            s(:lvar, :e), :message)),
        s(:send, nil, :warn,
          s(:str, "Run `bundle install` to install missing gems")),
        s(:send, nil, :exit,
          s(:send,
            s(:lvar, :e), :status_code)))), nil)):RuboCop::AST::Node
```

**Comment by ydah** [comment]
Oops, sorry. Thank you for suggestion. This PR has been updated.

**Comment by koic** [comment]
Thanks!

**Pull request merged**